bee-message/src/output/mod.rs

```diff
@@ -199,6 +202,50 @@ impl Output {
             Self::Nft(output) => Some(output.immutable_feature_blocks()),
         }
     }
+
+    /// Checks if a sufficient storage deposit was made for the given [`Output`].
+    pub fn check_sufficient_storage_deposit(&self, config: &ByteCostConfig) -> Result<(), Error> {
+        let maybe_storage_deposit_condition = match self {
+            #[cfg(feature = "cpt2")]
+            Output::SignatureLockedSingle(_) => return Ok(()),
+            #[cfg(feature = "cpt2")]
+            Output::SignatureLockedDustAllowance(_) => return Ok(()),
+            Output::Treasury(_) => return Ok(()), // `TreasureOutput`s don't have `UnlockConditions`.
+            Output::Basic(output) => output
+                .unlock_conditions()
+                .get(StorageDepositReturnUnlockCondition::KIND),
+            Output::Alias(output) => output
+                .unlock_conditions()
+                .get(StorageDepositReturnUnlockCondition::KIND),
+            Output::Foundry(output) => output
+                .unlock_conditions()
+                .get(StorageDepositReturnUnlockCondition::KIND),
+            Output::Nft(output) => output
+                .unlock_conditions()
+                .get(StorageDepositReturnUnlockCondition::KIND),
+        };
+
+        let required = minimum_storage_deposit(config, self);
+
+        if let Some(UnlockCondition::StorageDepositReturn(storage_deposit_condition)) = maybe_storage_deposit_condition
+        {
+            let deposit = storage_deposit_condition.amount();
+            if deposit >= required {
```

Comment:
Why not
```suggestion
            if storage_deposit_condition.amount() >= required {
```
? Then it's similar to below